src/commands/music/index.js

```diff
@@ -0,0 +1,215 @@
+      channel[0].join();
+      timer = setTimeout(vleave, 1200000, client, evt);
+    } else {
+      return Promise.resolve(T('music_not_found', lang));
+    }
+  } else {
+    const channel = evt.message.guild.voiceChannels.find(c => c.name.toLowerCase() === suffix.toLowerCase());
+    if (channel) {
+      channel.join();
+      timer = setTimeout(vleave, 1200000, client, evt);
+    } else {
+      return Promise.resolve(T('music_not_found', lang));
+    }
+  }
+}
+
+function play(client, evt, music, lang) {
+  const info = client.VoiceConnections.getForGuild(evt.message.guild);
+  if (!info) return Promise.resolve(T('music_not_connected', lang));
+  const encoder = info.voiceConnection.createExternalEncoder({
+    type: 'ffmpeg',
+    realtime: true,
+    source: music.url,
+    outputArgs: ['-compression_level', 7]
+  });
+  clearTimeout(timer);
+  encoder.play();
+  encoder.once('end', () => {
+    delSong(evt.message.guild.id);
+    getSong(evt.message.guild.id, (err, song) => {
+      if (err) throw err;
+      if (song) {
+        play(client, evt, JSON.parse(song));
+      } else {
+        timer = setTimeout(vleave, 1200000, client, evt);
+        return Promise.resolve(T('music_queue_empty', lang));
+      }
+    });
+  });
+}
+
+function request(client, evt, suffix, lang) {
+  if (!suffix) return Promise.resolve(T('request_usage', lang));
+  const info = client.VoiceConnections.getForGuild(evt.message.guild);
+  if (!info) return Promise.resolve(T('music_not_connected', lang));
+  if (suffix.indexOf('youtube.com/playlist') !== -1) return Promise.resolve('Playlist support coming soon');
+  return ytdl.getInfo(suffix, ['-f', 'bestaudio'], (err, media) => {
+    if (err) throw err;
+    if (!media) return T('request_error', lang);
+    if (media.length_seconds > 5400) return T('request_length', lang);
+    const formats = media.formats.filter(f => f.container === 'webm').sort((a, b) => b.audioBitrate - a.audioBitrate);
+    const bestaudio = formats.find(f => f.audioBitrate > 0 && !f.bitrate) || formats.find(f => f.audioBitrate > 0);
+    if (!bestaudio) return T('request_unavailable', lang);
+    const music = {
+      user: evt.message.author.username,
+      title: media.title,
+      duration: media.length_seconds,
+      url: bestaudio.url
+    };
+    getSong(evt.message.guild.id, (err, song) => {
+      if (err) throw err;
+      if (!song) {
+        addSong(evt.message.guild.id, music);
+        play(client, evt, music);
+      } else {
+        addSong(evt.message.guild.id, music);
+      }
+    });
+    return Promise.resolve(`\`${media.title}\` ${T('music_queue_added', lang)}`);
+  });
+}
+
+function stop(client, evt, lang) {
+  const info = client.VoiceConnections.getForGuild(evt.message.guild);
+  if (info) {
+    timer = setTimeout(vleave, 1200000, client, evt);
+    info.voiceConnection.getEncoderStream().unpipeAll();
+    delSongs(evt.message.guild.id);
+    delSong(evt.message.guild.id);
```

Comment:
Do you have to run both of these? The function name makes it sound like just running `delSonds` would be more then enough.